From 263736340604c89d81ae3ec8a6ab9a31bf40e60b Mon Sep 17 00:00:00 2001
From: Anton Mikanovich <amikan@ilbers.de>
Date: Tue, 27 Jul 2021 12:04:54 +0000
Subject: [PATCH 7/7] ubifs: Fix output files permissions

Need to change imager output files owner to the current user to make
them available outside Isar without sudo.

Signed-off-by: Anton Mikanovich <amikan@ilbers.de>
---
 meta/classes/fit-img.bbclass   | 1 +
 meta/classes/ubi-img.bbclass   | 1 +
 meta/classes/ubifs-img.bbclass | 1 +
 3 files changed, 3 insertions(+)

diff --git a/meta/classes/fit-img.bbclass b/meta/classes/fit-img.bbclass
index da8e1da..98ed8eb 100644
--- a/meta/classes/fit-img.bbclass
+++ b/meta/classes/fit-img.bbclass
@@ -23,6 +23,7 @@ do_fit_image() {
     # Create fit image using buildchroot tools
     sudo chroot ${BUILDCHROOT_DIR} /usr/bin/mkimage ${MKIMAGE_ARGS} \
                 -f '${PP_WORK}/${FIT_IMAGE_SOURCE}' '${PP_DEPLOY}/${FIT_IMAGE_FILE}'
+    sudo chown $(id -u):$(id -g) '${DEPLOY_DIR_IMAGE}/${FIT_IMAGE_FILE}'
 
     image_undo_mounts
 }
diff --git a/meta/classes/ubi-img.bbclass b/meta/classes/ubi-img.bbclass
index 0f4c4cb..5ce7686 100644
--- a/meta/classes/ubi-img.bbclass
+++ b/meta/classes/ubi-img.bbclass
@@ -26,6 +26,7 @@ do_ubi_image() {
     # Create ubi image using buildchroot tools
     sudo chroot ${BUILDCHROOT_DIR} /usr/sbin/ubinize ${UBINIZE_ARGS} \
                 -o '${PP_DEPLOY}/${UBI_IMAGE_FILE}' '${PP_WORK}/${UBINIZE_CFG}'
+    sudo chown $(id -u):$(id -g) '${DEPLOY_DIR_IMAGE}/${UBI_IMAGE_FILE}'
 
     image_undo_mounts
 }
diff --git a/meta/classes/ubifs-img.bbclass b/meta/classes/ubifs-img.bbclass
index 7d78a79..74a167a 100644
--- a/meta/classes/ubifs-img.bbclass
+++ b/meta/classes/ubifs-img.bbclass
@@ -25,6 +25,7 @@ do_ubifs_image() {
     # Create ubifs image using buildchroot tools
     sudo chroot ${BUILDCHROOT_DIR} /usr/sbin/mkfs.ubifs ${MKUBIFS_ARGS} \
                 -r '${PP_ROOTFS}' '${PP_DEPLOY}/${UBIFS_IMAGE_FILE}'
+    sudo chown $(id -u):$(id -g) '${DEPLOY_DIR_IMAGE}/${UBIFS_IMAGE_FILE}'
 
     image_undo_mounts
 }
-- 
2.20.1

